### PR TITLE
fix: Selection of breakdown type for feature flags

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

For funnels, the breakdown does not work for feature flags as it specifies the wrong type

## Changes

* Quick fix on the frontend to enforce the correct breakdown type.

Ideally this would surface more aggressively by the backend if the type is unsupported...

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

* Created a non-working funnel locally, fixed and saw that after breakdowns worked

### CODE REVIEW

**Review by pauldambra — APPROVED** [review]
introduced in #10668 for future context

LGTM ⛴

**Comment by neilkakkar** [comment]
Atleast one more place where this is borked: When filtering for a feature flag, the autocomplete doesn't work anymore, because it tries to go to:

`https://app.posthog.com/api/feature/values/?key=$feature/3275-formulas` , instead of `https://app.posthog.com/api/event/values/?key=$feature/3275-formulas`

**Comment by neilkakkar** [comment]
And another: When filtering trends/funnels by a feature flag value, the property `type` sent is `feature`, instead of `event`.

Just highlighting here, mind confirming that this PR fixes both these cases as well?

**Review by neilkakkar — APPROVED** [review]
Just confirmed, uses the same codepath

**Comment by camerondeleone** [comment]
If we’re making changes here, I noticed that the recent pr doesn’t really address the main issue it was trying to solve for (which was, IMO, to declutter events by removing all the ff-related stuff to a specific tab). eg we still have two of the three ff event properties still showing up under `Event Properties`. 

An ideal solution here would be to move _all_ feature flag related properties to the `Feature Flags` tab, and probably to only leave two of those properties visible (ie do not show the others, bc they are too noisy and there is no real-world use case for filtering/breaking down on them that I can think of).

Basically, there are two questions that need answering here:

- show me x analysis grouped/filtered by variant of the ff
- show me x analysis grouped/filtered by presence/absence of the flag

We are showing a bunch of repetitive info (some nested json where you would need to use `contains` to search for the flag or the `true` enabled)

Actually, I really think we should hide everything ff-related that's not currently showing up in the newly created `Feature Flags` tab, but open to other interpretations there.






image

**Comment by benjackwhite** [comment]
@camerondeleone @neilkakkar  can it be that these are some sort of old legacy values that aren't actually reflected for most customers? Just noticed that none of the `$active_feature_flags__N` don't show for me on any other account than Posthog.

**Comment by neilkakkar** [comment]
@benjackwhite it's a result of 2 things:

1. We have `$active_feature_flags` property, which is an array of active ffs.
2. We have a property flattener plugin, which converts all array properties into custom properties like: `$active_feature_flags__0` with numbers from 0 to N. (Some people might have this enabled, others might not). And N depends on max number of enabled flags.

**Comment by benjackwhite** [comment]
Ah thanks for the context @neilkakkar . This makes it a little trickier then I guess. We could build in custom logic to handle the property flattener plugin although that feels a bit messy, modifying the core based on a plugin?

**Comment by camerondeleone** [comment]
Ah, didn't even think about that. If this shows up only for us @benjackwhite I can live with it.

**Comment by neilkakkar** [comment]
Yeah, and feels like doing so would defeat the purpose of the plugin? It exists to have access to nested json / arrays. But removing that seems counter to what its for. We could probably get rid of the plugin instead, if it's not really being used 🤷 .Or mod it to not flatten specific properties.

**Comment by benjackwhite** [comment]
Either way merging this 👍
